mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -163,12 +163,20 @@ class RFormula @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     }.toMap
 
     // Then we handle one-hot encoding and interactions between terms.
+    var hasReferenceCategory = false
```

Comment:
I think this is just used as an indicator whether a `dropLast(false)` has been set. 
However, I do think we can have better name than `hasReferenceCategory`.
Basically what's needed is just set `dropLast(false)` for the first string. 
How about something like this:
```
var firstString = true 
if (!hasIntercept && firstString) {
          encoder = encoder.setDropLast(false)
          firstString = false
}
```